File: jobboard/src/main/java/com/epitech/jobboard/Services/AdvertissementService.java
package com.epitech.jobboard.Services;

import java.util.List;

import javax.transaction.Transactional;

import com.epitech.jobboard.Entities.Advertissement;


public interface AdvertissementService {
    /**
     * Creates a new Advertissement.
     * @param advert The Advertissement to create
     */
    void create(Advertissement advert);

    /**
     * Updates an advertisement.
     *
     * @param  advert  the advertisement to be updated
     */
    @Transactional
    void update(Advertissement advert);
    
    /**
     * Deletes an advertisement.
     * @param id The ID of the advertisement to delete
     */
    void deleteById(int id);

    /**
     * Retrieves an advertisement by its ID.
     * @param id The ID of the advertisement
     * @return The advertisement with the specified ID
     */
    Advertissement findById(int id);

    /**
     * Retrieves all advertisements
     * @return A list of all advertisements
     */
    List<Advertissement> findAll();
}

File: project-jobboard/src/Services/AdvertissementService.js
import axios from 'axios';

axios.defaults.baseURL = 'http://localhost:8080';

axios.interceptors.request.use(
    (config) => {
      const token = localStorage.getItem('userToken');
      if (token) {
        config.headers['Authorization'] = `Bearer ${token}`;
      }
      return config;
    },
    (error) => {
      return Promise.reject(error);
    }
  );

/**
 * Gets all advertisements from the API
 * 
 * @return {Promise} A promise that resolves to an array of advertisements
 */
function getAdvertissement () {
    return axios.get('/api/advertissements');
}

/**
 * Retrieves an advertisement by ID.
 *
 * @param {number} id - The ID of the advertisement.
 * @return {Promise} A promise that resolves to the advertisement data.
 */
function getAdvertissementById (id) {
    return axios.get(`/api/advertissements/${id}`);
}

/**
 * Creates an adverstisement.
 *
 * @param {Object} advertissement - The adverstisement object to be created.
 * @return {Promise} A promise that resolves to the created adverstisement.
 */
function createAdvertissement (advertissement) {
    return axios.post('/api/advertissements', advertissement);
}

/**
 * Updates an advertisement.
 *
 * @param {Object} advertissement - The advertisement to be updated.
 * @return {Promise} - A promise that resolves with the updated advertisement.
 */
function updateAdvertissement (advertissement) {
    return axios.put(`/api/advertissements/`, advertissement);
}

/**
 * Deletes an advertisement with the given ID.
 *
 * @param {number} id - The ID of the advertisement to delete.
 * @return {Promise} A promise that resolves to the result of the delete operation.
 */
function deleteAdvertissement (id) {
    return axios.delete(`/api/advertissements/${id}`);
}

const AdvertissementService = {
    getAdvertissement,
    getAdvertissementById,
    createAdvertissement,
    updateAdvertissement,
    deleteAdvertissement
};

export default AdvertissementService;
